### ISSUE TITLE
WPA2-Enterprise support ??

### ISSUE THREAD

**Issue by one0410** [adafruit/circuitpython#7083]
Hi,
I found that there is no support for WPA2-Enterprise currently.
I really need this to connect to my customers enterprise network.
Do you have a plan to support it ??

**Comment by dhalbert** [comment]
Which chip family do you want this on?

**Comment by one0410** [comment]
I'm sorry for that I didn't mention for using ESP32-S3 currently.

**Comment by MaxMyzer** [comment]
Any chance of Pi Pico W (CYW_43) support?

**Comment by dhalbert** [comment]
We depend on https://github.com/georgerobotics/cyw43-driver for the Pico W, so we'd await that. It's not clear to me whether just defining some more authmode constants will work or it needs further support.

**Comment by one0410** [comment]
Hi, any update for support enterprise wifi for esp32 ??

**Comment by anecdata** [comment]
Looks like different methods have different levels of implementation (and use) complexity
https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-guides/wifi-security.html#wi-fi-enterprise
Which method(s) are needed?

**Comment by one0410** [comment]
> Looks like different methods have different levels of implementation (and use) complexity https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-guides/wifi-security.html#wi-fi-enterprise Which method(s) are needed?

I think that most company will use PEAP and EAP-TTLS most. These methods can be used with username & password.
Could you help to implement these methods first? Thank you very much.

**Comment by anecdata** [comment]
I was just clarifying requirements. I'm not the one to do the implementation. I've never used enterprise and have no access to test.

**Comment by one0410** [comment]
Thanks for your quickly response.
So do you know is there any plans or possibilities to implement these recently?

**Comment by dhalbert** [comment]
We have no immediate plans, but it doesn't mean we don't want to. It's not so easy to come up with a test situation at home.

**Comment by one0410** [comment]
Got it.
Wish we can have these features in near future when you team have time.
Thank you again.

**Comment by mlamparter** [comment]
Just checking in to see if there has been any progress on this.  I have contributed zero to this project so I am incredibly grateful for the hard work others have put into.  Just want to make it clear I'm respectfully checking in, not making a demand :)

Not sure if it matters or offers any help, but it looks like MicroPython may provide support for this in their [upcoming 1.28 release on March 9](https://github.com/micropython/micropython/pull/17234).

I work at a university, so if having access to a test bed would be helpful for this feature I may be able to offer some assistance.

**Comment by h-milz** [comment]
@mlamparter @dhalbert  This is correct. I'm just finalizing my PR and it is due in MPY 1.28. Please note it is ESP32 only at the moment (mostly a wrapper for the existing ESP-IDF 5.5 functions). The main commit is just an extension to wlan.connect() and should build cleanly on Circuitpython as well. Feel free to give it a test and let me know if something breaks. In this case I may be able to fix a thing or two before March 9.
